Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -2326,6 +2326,11 @@ bool LocalFrameView::scrollToFragment(const URL& url)
             
             if (highlightRanges.size()) {
                 auto range = highlightRanges.first();
+                auto* commonAncestor = commonInclusiveAncestor<ComposedTree>(range);
+                while (commonAncestor && !is<Element>(commonAncestor))
+                    commonAncestor = commonAncestor->parentElement();
+                if (commonAncestor)
+                    document->setCSSTarget(downcast<Element>(commonAncestor));
```

Comment:
I can't find where the spec says to set the `:target` pseudo class state on the common ancestor. Can you clarify where that is done?

Reply:
Sure, see footnote [1], then the section "In scroll to the fragment, handle a indicated part that is a range....", there is step 3.6 which sets the target.
It is a bit hard to refer to this in the commit message since there are multiple mockeypatches for § 7.4.6.3.

Reply:
Ah right, thanks